### PR TITLE
Improve balance tracking

### PR DESCRIPTION
**Pull request by ben-kaufman**

### CODE REVIEW

File: src/mappings/GenericScheme/mapping.ts [hunk 1]
```diff
@@ -47,6 +48,9 @@ export function handleProposalExecuted(
   if (ent != null) {
     ent.executed = true;
     ent.returnValue = event.params._genericCallReturnValue;
+    if (BigInt.fromI32(0).lt(ent.value)) {
```
Comment (orenyodfat) [inline src/mappings/GenericScheme/mapping.ts:51]: should it check if the call succeed?

Comment (ben-kaufman) [reply src/mappings/GenericScheme/mapping.ts:51]: Well if the call fails the event won't happen in the first place. So I don't think we need to/ can do that.

Comment (orenyodfat) [reply src/mappings/GenericScheme/mapping.ts:51]: not sure about that as if https://github.com/daostack/arc/blob/master/contracts/controller/Avatar.sol#L58 revert the event will still be emitted

Comment (ben-kaufman) [reply src/mappings/GenericScheme/mapping.ts:51]: No, we check here before calling the Executed event that the call succeeded https://github.com/daostack/arc/blob/1d95202c3d2c2daef7c8dffcf7935e236c15c24a/contracts/schemes/GenericScheme.sol#L114

Comment (orenyodfat) [reply src/mappings/GenericScheme/mapping.ts:51]: I think this should be handle at the Avatar module listen to the handle GenericCall event.
not in the genericScheme

Comment (ben-kaufman) [reply src/mappings/GenericScheme/mapping.ts]: Ok, I changed it

File: src/mappings/Avatar/mapping.ts [hunk 2]
```diff
@@ -27,6 +28,12 @@ function handleAvatarBalance(
   store.set('AvatarContract', avatar.id, avatar);
 }
 
+export function handleGenericCall(event: GenericCall): void {
+  if (event.params._success) {
```
Comment (orenyodfat) [inline src/mappings/Avatar/mapping.ts:32]: should it also check if the balance is not 0 ?

**Pull request merged**